### ISSUE TITLE
move-text-up moves point too far

### ISSUE THREAD

**Issue by Wilfred** [emacsfodder/move-text#2]
Given the following text in a buffer, where | is point (note final blank line):

```
first
second
|third

```

`move-text-up` produces the following:

```
|first
third
second

```

whereas the expected behaviour is:

```
first
|third
second

```

I have reproduced this on both Emacs 24.5 and Emacs trunk, so I assumed the version check (for 24.3+) in the code was simply incorrect.

Note that behaviour is slightly different if the current line is the last line in the buffer. Before:

```

first
second
|third
```

after:

```
|
second
first
third
```

**Comment by jasonm23** [comment]
Does a blank or non blank follow the selection?

**Comment by Wilfred** [comment]
Sorry, I don't understand. The region isn't active, if that's what you mean.

**Comment by jasonm23** [comment]
Right now is not good, I'm on a packed MRT Train in Bangkok.

Tomorrow :)

**Comment by Wilfred** [comment]
Interestingly, it looks like @purcell was involved in the Emacs 24.3 compatibility patch: https://www.emacswiki.org/emacs/Comments_on_move-text.el -- I don't suppose you recall the issue Steve?

**Comment by purcell** [comment]
I don't recall the details, but IIRC one of the functions involved left the point in a different place in the different Emacs versions. These days I use the `move-dup` package instead.

**Comment by purcell** [comment]
Ah yes, actually it was probably the last-line-in-file thing.

**Comment by jasonm23** [comment]
@Wilfred is it possible for you to do a screen recording to demonstrate the problem? Or can you confirm if it's the problem as described by @purcell?

**Comment by iquabius** [comment]
I'm on GNU Emacs 25.1.50.2, and I'm also experiencing the same bug.

The result of repeatedly pressing M-up is depicted in the following gif:

![move-text-bug 2 optimised](https://cloud.githubusercontent.com/assets/3139777/16594740/c55c732c-42b9-11e6-8362-3e65a67d29b3.gif)

**Comment by jasonm23** [comment]
@iquabius please try testing from an `emacs -Q` startup.

in `*scratch*` you can do:

```
(package-initialize)
(require 'move-text)
```

and then eval buffer.

Try using move text in your scratch, see if you get the same issue.

We'll need to rule out all your other packages.

**Comment by jasonm23** [comment]
BTW @iquabius your issue is different from @Wilfred's

**Comment by jasonm23** [comment]
Fixed in 2.0x
